### PR TITLE
Add SPDX header

### PR DESCRIPTION
**Pull request by AndyScherzinger**
* Adding SPDX headers
* Adding a GH action to check for compliance
* Adding a status badge for REUSE compliance
* Replaced most icons with current Material Design ones
* Replaced most logos when available with the originals

### CODE REVIEW

File: LICENSES/MIT.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+MIT License
+
+<year> <copyright holders>
```
Comment (hamza221) [inline LICENSES/MIT.txt:3]: Not sure if it's left blank on purpose
```suggestion
<2024> <Nextcloud GmbH and Nextcloud contributors>
```

Comment (AndyScherzinger) [reply LICENSES/MIT.txt:3]: Yes, left blank on purpose 👍

File: .github/workflows/lint-php.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
       php-versions: ${{ steps.versions.outputs.php-versions }}
     steps:
       - name: Checkout app
-        uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+        uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
```
[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply .github/workflows/lint-php.yml:27]: Sure, I can revert them or renovate does it's job and we rebase?

File: LICENSES/MIT.txt (hunk 1 above)
Comment (st3iny) [reply LICENSES/MIT.txt:3]: I think it's blank on purpose. The authors of the corresponding SPDX headers are supposed to be substituted.

**Review by st3iny — APPROVED** [review]
Thanks Andy :rocket:

File: .github/workflows/lint-php.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (st3iny) [reply .github/workflows/lint-php.yml:27]: Revert them here and Renovate will do its thing in another PR.

[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply .github/workflows/lint-php.yml:27]: How to do that without the current git commit history?

[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply .github/workflows/lint-php.yml:27]: Meaning, yes, but I would actually do a revert commit for the files in question

[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply .github/workflows/lint-php.yml]: Resolved, by asking renovate for a helping hand 👍

**Comment by AndyScherzinger** [comment]
Thanks for the reviews 👍

**Pull request merged**